Assets/UI/Panels/citypaneloverview.lua

```diff
@@ -173,6 +178,88 @@ function EnsureDesiredLens()
 end
 -- ==== CQUI CUSTOMIZATION END ======================================================================================== --
 
+
+-- ==== CQUI CUSTOMIZATION BEGIN ======================================================================================== --
+-- Infixo Great Works
+
+local tSlotTypes:table = {};
+
+function InitializeSlotTypes()
+    for row in GameInfo.GreatWork_ValidSubTypes() do
+        local sGWSlotType:string = row.GreatWorkSlotType;
+        if tSlotTypes[sGWSlotType] == nil then
+            tSlotTypes[sGWSlotType] = { Num = 0, Icons = "", IconsTT = "" };
+        end
+        local slotType:table = tSlotTypes[sGWSlotType];
+        slotType.Num = slotType.Num + 1;
+        local gwObject:table = GameInfo.GreatWorkObjectTypes[ row.GreatWorkObjectType ];
+        slotType.Icons = slotType.Icons..gwObject.IconString;
+        slotType.IconsTT = slotType.IconsTT..(slotType.IconsTT == "" and "" or "[NEWLINE]")..gwObject.IconString..Locale.Lookup(gwObject.Name);
```

Comment:
I am curious what this does - I've been looking at it for a good 30 seconds trying to figure out the order of operations.  does it always add a newline?

Reply:
It adds a Newline tag to IconsTT only when it is not empty. When it is empty, then nothing extra is added. That way the ToolTip is properly formatted, e.g. line1 newline line2 newline line3.